src/SlnGen.Build.Tasks/Internal/SlnFile.cs

```diff
@@ -122,27 +110,33 @@ public void Save(TextWriter writer)
             writer.WriteLine("Global");
 
             writer.WriteLine("	GlobalSection(SolutionConfigurationPlatforms) = preSolution");
-            foreach (string configuration in _configurations)
+
+            IEnumerable<string> globalConfigurations = new HashSet<string>(_projects.SelectMany(p => p.Configurations)).Distinct();
```

Comment:
I know VS has logic to parse property groups like this:
```xml
<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
```

Do you think its worth parsing conditions like that to see what the project is configured against?

Reply:
This is actually how it is originally parsed. This is read from ConditiondProperties. See the extension method GetConditionedPropertyValuesOrDefault

Reply:
https://github.com/Microsoft/msbuild/blob/03d1435c95e6a85fbf949f94958e743bc44c4186/src/Build/Definition/Project.cs#L706